### ISSUE TITLE
Make ListPipeline streaming & parallelized

### ISSUE THREAD

**Issue by msteffen** [pachyderm/pachyderm#3895]
Currently `ListPipeline`, unlike `ListJob`, `ListCommit`, etc is not a streaming RPC. This can make the call frustrating for users who have a lot of pipelines, since it can hang for a long time and then return a complete list of your pipelines at the end.

- [ ] We should make ListPipeline a streaming RPC

Also, once ListPipeline is streaming, we should refactor it so that
- [ ] By default it doesn't make any PFS RPCs (and doesn't show the `input` column in the `pachctl` output)
- [ ] If it does make PFS RPCs (to read pipeline specs), then those are parallelized.

**Comment by ysimonson** [comment]
`ListJob` and `ListCommit` have `ListJobStream` and `ListCommitStream` for streaming variants, no?

**Comment by msteffen** [comment]
Correct. We just never added a ListPipelineStream variant of ListPipeline.

**Comment by ysimonson** [comment]
^ that's probably a reply to a comment I left but deleted once I realized I had misread the ticket
